core/src/main/scala/kafka/security/authorizer/AclAuthorizer.scala

```diff
@@ -63,9 +64,14 @@ object AclAuthorizer {
     def exists: Boolean = zkVersion != ZkVersion.UnknownVersion
   }
 
-  class AclSeqs(classes: Seq[AclEntry]*) {
-    def find(p: AclEntry => Boolean): Option[AclEntry] = classes.flatMap(_.find(p)).headOption
-    def isEmpty: Boolean = !classes.exists(_.nonEmpty)
+  class AclSeqs(seqs: Seq[AclEntry]*) {
+    def find(p: AclEntry => Boolean): Option[AclEntry] = {
```

Comment:
Should it need comment to remind reader that this style is for optimization.

Reply:
I think this is obvious, no? `find` should generally short-circuit and not go through all the items. That's how it works for all collection implementations

I think this kind of comment makes sense in `matchingAcls` where I added one.

Reply:
@chia7712 I looked at the code again and I guess the intent may not be clear. I added a comment.